### PR TITLE
gcode: M30 - Properly handle absolute paths and relative paths

### PR DESCRIPTION
**Pull request by bkerler**
This fixes issue #5005.

The M30 command (delete file on SD/USB) was failing because it didn't properly handle:

- Leading whitespace in the filename argument
- Simplify3D format (file size appended after space)
- Absolute paths that already include /usb/ or /sd/ prefix

### CODE REVIEW

File: src/marlin_stubs/sdcard/M20-M30_M32-M34.cpp [hunk 1]
```diff
@@ -199,14 +200,41 @@ void GcodeSuite::M29() {
  *    M30 [filename]
  */
 void GcodeSuite::M30() {
+    // Skip leading whitespace
+    const char *filename = parser.string_arg;
```
Comment (ttabi) [inline src/marlin_stubs/sdcard/M20-M30_M32-M34.cpp:203]: Do you need to test for NULL first?

File: src/marlin_stubs/sdcard/M20-M30_M32-M34.cpp [hunk 2]
```diff
@@ -199,14 +200,41 @@ void GcodeSuite::M29() {
...
+    while (*filename == ' ') {
+        filename++;
+    }
+
+    // Handle Simplify3D format which includes file size after a space (same as M23)
+    for (char *fn = const_cast<char *>(filename); *fn; ++fn) {
+        if (*fn == ' ') {
```
Comment (ttabi) [inline src/marlin_stubs/sdcard/M20-M30_M32-M34.cpp:219]: Does the firmware handle escape characters?  That is, if I type in hello\ world.gcode, will it recognize it the entire filename?  Does the firmware allow the user to upload files with spaces in their names?

File: src/marlin_stubs/sdcard/M20-M30_M32-M34.cpp [hunk 3]
```diff
@@ -199,14 +200,41 @@ void GcodeSuite::M29() {
...
+            *fn = '\0';
+            break;
+        }
+    }
+
+    // Skip empty filename
+    if (!filename || !*filename) {
```
Comment (ttabi) [inline src/marlin_stubs/sdcard/M20-M30_M32-M34.cpp]: The !filename test is too late here.

File: src/marlin_stubs/sdcard/M20-M30_M32-M34.cpp [hunk 4]
```diff
@@ -199,14 +200,41 @@ void GcodeSuite::M29() {
...
+        SERIAL_ECHOLNPGM("Deletion failed: No filename provided");
+        return;
+    }
+
     ArrayStringBuilder<FF_MAX_LFN> filepath;
-    filepath.append_printf("/usb/%s", parser.string_arg);
+
+    // Handle paths that may already have /usb/ or /sd/ prefix
+    if (strncmp(filename, "/usb/", 5) == 0 || strncmp(filename, "/sd/", 4) == 0) {
+        filepath.append_string(filename);
```
Comment (ttabi) [inline src/marlin_stubs/sdcard/M20-M30_M32-M34.cpp:235]: You need to test whether the filename is too long to fit into filepath.

Comment (bkerler) [reply src/marlin_stubs/sdcard/M20-M30_M32-M34.cpp:235]: ArrayStringBuilder already tracks overflows, but yeah, will check the is_ok function.

File: src/marlin_stubs/sdcard/M20-M30_M32-M34.cpp (hunk 1 above)
Comment (bkerler) [reply src/marlin_stubs/sdcard/M20-M30_M32-M34.cpp:203]: Right.

File: src/marlin_stubs/sdcard/M20-M30_M32-M34.cpp (hunk 3 above)
Comment (bkerler) [reply src/marlin_stubs/sdcard/M20-M30_M32-M34.cpp]: if we need to check for nullptr, then yes

File: src/marlin_stubs/sdcard/M20-M30_M32-M34.cpp (hunk 2 above)
Comment (bkerler) [reply src/marlin_stubs/sdcard/M20-M30_M32-M34.cpp:219]: The gcode parser has no escape character mechanism. The truncation-at-space is intentional — it mirrors what M23 does for Simplify3D compatibility (where the host appends a file size, e.g. M30 filename.gcode 12345).
